recOrder/scripts/cli.py

```diff
@@ -1,148 +1,181 @@
+from iohub.reader_base import ReaderBase
+from iohub.ngff import NGFFNode, Plate, Position, TiledPosition
+
+
+def _get_reader(filename):
+    fmt, extra_info = _infer_format(filename)
+    if fmt == "omezarr" and extra_info == "0.4":
+        reader = open_ome_zarr(filename, mode="r")
+    else:
+        reader = read_micromanager(filename, data_type=fmt)
+    return reader
+
+
+def _build_complete_position_list(reader):
+    if isinstance(reader, ReaderBase):
+        positions = range(reader.get_num_positions())
+    elif isinstance(reader, Plate):
+        positions = [x[0] for x in list(reader.positions())]
+    elif isinstance(reader, Position) or isinstance(reader, TiledPosition):
+        positions = ["0"]
+    else:
+        raise (
+            NotImplementedError,
+            f"`recOrder view` does not support {type(reader)}.",
+        )
+    return positions
 
 
-@click.group()
-def cli():
-    print(
-        "\033[92mrecOrder: Computational Toolkit for Label-Free Imaging\033[0m\n"
-    )
-
+def _build_array_list(reader, positions, layers):
+    arrays = []
+    if isinstance(reader, ReaderBase):
+        positions = [int(x) for x in positions]
 
-@cli.command()
-@click.help_option("-h", "--help")
-@click.argument("filename")
-def info(filename):
-    """View a dataset's basic metadata"""
-    print(f"Reading file:\t {filename}")
+        # Positions as layers
+        if layers == "position" or layers == "p":
+            for position in positions:
+                arrays.append(reader.get_zarr(position))
 
-    from waveorder.io import WaveorderReader
+        # Channels as layers
+        elif layers == "channel" or layers == "c":
+            print(
+                """WARNING: for sending channels to layers is more expensive than 
+sending positions to layers. Try loading a small number of positions."""
+            )
 
-    reader = WaveorderReader(filename)
-    print_reader_info(reader)
+            ptzyx = (len(positions),) + (reader.shape[0],) + reader.shape[2:]
+            for channel in range(int(reader.channels)):
+                temp_data = np.zeros(ptzyx)
+                for k, position in enumerate(positions):
+                    temp_data[k] = reader.get_array(position)[:, channel, ...]
+                arrays.append(temp_data)
+
+    elif isinstance(reader, NGFFNode):
+
+        # Positions as layers
+        if layers == "position" or layers == "p":
+            for position in positions:
+                arrays.append(reader[position].data)
+
+        # Channels as layers
+        elif layers == "channel" or layers == "c":
+            T, C, Z, Y, X = reader[positions[0]].data.shape
+            ptzyx = (len(positions),) + (T, Z, Y, X)
+            for c in range(C):
+                temp_data = np.zeros(ptzyx)
+                for k, position in enumerate(positions):
+                    temp_data[k] = reader[position].data[:, c, ...]
+                arrays.append(temp_data)
```

Comment:
@ziw-liu I could use your help here. This naive solution loads the data into RAM then displays it. Can you suggest a solution that will lazily load?

Reply:
napari accepts dask arrays as layer data. Wrapping each zarr array in dask and stacking them should acheive lazy loading.

Reply:
For now I'm "resolving" this by adding a TODO...for now it's nice to have but not needed.